app/src/org/odk/collect/android/utilities/GeoUtils.java

```diff
@@ -79,7 +80,11 @@ public static void showNoGpsDialog(Activity activity,
             factory.makeCancelable();
             factory.setOnCancelListener(onCancel);
         }
-        factory.showDialog();
+        if (activity instanceof CommCareActivity) {
```

Comment:
Should probably be `DialogController` here too, then we can implement the appropriate methods in `GeoPointActivity` and dialog persistence will work.